quipucords/api/models.py

```diff
@@ -53,3 +53,16 @@ def update(self, request, *args, **kwargs):
 
     class Meta:
         verbose_name_plural = 'Host Credentials'
+
+
+class NetworkProfile(models.Model):
+    """A network profile connects a list of credentials and a list of hosts."""
+    name = models.CharField(max_length=64, unique=True)
+    # Comma-separated list of host specifiers
+    hosts = models.CharField(max_length=1024)
+    ssh_port = models.IntegerField()
+    # Comma-separated list of credential names. We identify
+    # credentials by name, not ID, to support the behavior where a
+    # user creates a credential of a given name, deletes it, and makes
+    # a new one of the same name.
+    credentials = models.CharField(max_length=1024)
```

Comment:
Swagger doc had this as *credential_ids*. I think we should use an ID and and again some of the modeling relationships so we can consume an array here and we are referencing a stored Host Credential. Django should be doing some of the "existence" checking for us.

Reply:
I think it makes sense to use a relationship here. The side effect of recreating a credential with the same name can be worked around by taking advantage of an edit on the credential.